### PR TITLE
OpAMP Agent Extension

### PR DESCRIPTION
**Pull request by portertech**
Creating this PR for increased visibility, early feedback, and OpAMP server testing. This pull-request  supersedes https://github.com/SumoLogic/sumologic-otel-collector/pull/855.

This OpAMP agent implementation manages a single local configuration file (e.g. `opamp-remote-config.yaml`) intended to be consumed by the [glob configuration provider](https://github.com/SumoLogic/sumologic-otel-collector/tree/main/pkg/configprovider/globprovider). The configuration file directory is configurable via `remote_configuration_directory`, this directory must match the glob configuration provider pattern used by the collector (e.g. `remote_configuration_directory: /etc/otelcol-sumo/conf.d` -> `--config "glob:/etc/otelcol-sumo/conf.d/*"`). Remote configuration updates trigger a collector reload via a SIGHUP signal (to self).

Example OpAMP configuration for testing:

```
extensions:
  sumologic:
    install_token: ${SUMOLOGIC_INSTALL_TOKEN}
    api_base_url: "https://stag-open-events.sumologic.net"
  opamp:
    instance_uid: 01BX5ZZKBKACTAV9WEVGEMMVRZ
    endpoint: "wss://stag-opamp-events.sumologic.net/v1/opamp"
    remote_configuration_directory: /Users/sporter/Desktop/conf.d
```

```
./otelcolbuilder/cmd/otelcol-sumo --config /Users/sporter/Desktop/sumologic.yaml --config "glob:/Users/sporter/Desktop/conf.d/*"
```

### CODE REVIEW

**Comment by portertech** [comment]
This extension is currently able to authenticate to our OpAMP server and send a mock effective configuration on request.

```
2022-12-29T15:19:59.994-0800    info    opampextension@v0.0.0-00010101000000-000000000000/opamp_agent.go:91     Connected to the OpAMP server   {"kind": "extension", "name": "opamp"}
2022-12-29T15:19:59.995-0800    info    opampextension@v0.0.0-00010101000000-000000000000/opamp_agent.go:100    OpAMP server requested the effective configuration      {"kind": "extension", "name": "opamp"}
```

The OpAMP server does not yet reply with a remote configuration to apply. If it were to reply with one, the extension's OpAMP agent would produce a new effective configuration and report it to the server. The effective configuration is not yet applied to the collector itself (soon). This work is in a state where it is useful for testing the OpAMP server development.

**Review by sumo-drosiek — DISMISSED** [review]

**Comment by portertech** [comment]
@sumo-drosiek I'll definitely call upon you for a review once things stabilize 👍

**Review by perk-sumo — APPROVED** [review]
@portertech one thing that I thing we need is a README, other than that LGTM (as alpha feature that's disabled by default 😉 )

File: pkg/extension/opampextension/README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# OpAMP Agent Extension
```
Comment (portertech) [inline pkg/extension/opampextension/README.md:1]: @perk-sumo Is this sufficient? I used the Sumo extension readme as a base.

**Pull request merged**